src/xwayland/xwm.rs

```diff
@@ -561,6 +561,7 @@ impl X11WM {
                             None
                         },
                     });
+                    // TODO: If the window is not configured as part of this callback, we need to send a synthetic configure event
```

Comment:
Actually, I would suggest to *always* send a synthetic configure event after this. I learnt this the hard way: https://github.com/awesomeWM/awesome/commit/cd63cabaddabd61bbbea44bf4dc8a4f575901e92

(Yes, this commit adds dead code. `send_notice` will always end up being `true`. `client_send_configure()` is a helper function that sends a synthetic configure event.)

Don't ask me about the details, but I think I remember digging into Java's source code and finding some code that detects KWin and then behaves ICCCM conforming. There was a comment next to this claiming that this was a work-around for a bug in KWin. So... yeah, "mechanism not policy". Thanks, X11.

Reply:
If only there would be something better that could replace X11... ;-)

Sorry, I don't know such a list. My general rule of thumb is that I stopped debugging bug reports involving Java, because it always ended up being Java's fault and taking a lot of time.

For proper timestamps... yeah, well, X11 is no longer used (much) across high-latency network connections and so the kind of races that proper timestamps are supposed to avoid are no longer that important. And the spec only talks about "normal" clients. My "bad excuse" is that it does not directly apply to WMs. ;-)

I did eventually write https://github.com/awesomeWM/awesome/commit/ec63afc822 and https://github.com/awesomeWM/awesome/commit/ce814b4dbfbda5088955fb977555b4fcc0ff79fc to reduce the number of `XCB_CURRENT_TIME` uses. Dunno if that ever made any differences.

So.... dunno. Feel free to ask for help if some stuff behaves weirdly. And this "Java always wants a synthetic ConfigureNotify since it fails to properly track its position"-bug has become my pet peeve since I wasted sooo much time on debugging it back then.